pycloudlib/azure/cloud.py

```diff
@@ -863,6 +957,17 @@ def launch(
         # resources we are creating.
         self.tag = get_timestamped_tag(self.base_tag)
 
+        try:
+            self._check_compute_quota(instance_type)
+
+        except Exception as e:
+            # Clean up resources created during the failed attempt
+            self._log.error("Exception during launch: %s", e)
+            if self.resource_group:
```

Comment:
I think this block is not needed, as the class will clean up these resources using the `clean` function if instantiated as a context manager and leak them if not.